gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/AuthenticationService.java

```diff
@@ -377,6 +375,21 @@ public QueryResponse parseJwtToken(String jwtToken) {
         return fromCookie;
     }
 
+    /**
+     * Get the JWT token from the cookie to process the logout
+     *
+     * @param request the http request
+     * @throws TokenFormatNotValidException if the token format is not valid or the cookie is empty
+     * @return the JWT token
+     */
+    public Optional<String> getJwtTokenFromRequestToLogout(HttpServletRequest request) {
```

Comment:
Why new method? This TokenFormatNotValidException could be used also for  getJwtTokenFromRequest